src/js/App.js

```diff
@@ -1,70 +1,118 @@
-import React, { Component } from "react"
-import "../styles/App.css"
-import { STEPS } from "../config"
-import { Arrow } from "./components"
-import { Step } from "./pages"
+import React, { Component } from "react";
+import "../styles/App.css";
+import { STEPS } from "../config";
+import { Arrow, Button } from "./components";
+import { Step } from "./pages";
+
+const BUTTON_STEP = 2;
 
 class App extends Component {
   constructor(props, context) {
-    super(props, context)
+    super(props, context);
     this.state = {
-      activeStep: STEPS[0]
-    }
+      activeStep: STEPS[0],
+      steps: STEPS
+    };
+    this.onButtonClick = this.onButtonClick.bind(this);
   }
 
   render() {
     return (
       <div className="App">
-        <Arrow
-          direction="left"
-          handleOnClick={this.changeStep.bind(this, false)}
-        />
-        {STEPS.map((step, idx) => (
-          <div
-            className={this.getStepClass(step, idx)}
-            key={idx}
-          >
-            <Step
-              handleOnKeyDown={this.onKeyPressed}
-              step={step}
-            />
+        {this.getStepIdx() !== BUTTON_STEP && this.getStepIdx() && (
```

Comment:
@madisoncarr this conditional is returning the last value it reads, and JavaScript has a quirk where the number `0` is equal to `false`. So when you call `this.getStepIdx()` on the first step it returns `false` (which I think you're trying to do here). The issue is that on the first step (when idx is 0) it's not rendering the arrow, but is rendering the `0` returned from the second conditional.

You can fix this in at least two ways:
1. Do an explicit (and not implicit) true/false check:
```
this.getStepIdx() !== BUTTON_STEP && this.getStepIdx() !== 0 ...
```
2. Group the check for the button step and the index check and have that expression return `true` or `false`, which is the conditional check for a ternary operator that returns the arrow when `true`, or `null` when `false`.
```
{(this.getStepIdx() !== BUTTON_STEP && this.getStepIdx()) ? (
  <Arrow
    direction="left"
    handleOnClick={this.changeStep.bind(this, false)}
  />
) : null}
```

Reply:
@wade00 It does render a zero! Good catch!